### ISSUE TITLE
MueLu:  Potential typing mistake in BelosXpetraStatusTestGenResSubNorm (edited)

### ISSUE THREAD

**Issue by kliegeois** [trilinos/Trilinos#6082]
## Bug Report
@trilinos/muelu 
@tawiesn 

### Description
There is, from time to time, a seg fault in file [BelosXpetraStatusTestGenResSubNorm.hpp](https://github.com/trilinos/Trilinos/blob/develop/packages/muelu/adapters/belos/BelosXpetraStatusTestGenResSubNorm.hpp) in the block of lines 289-306:

```
    for (; p2<curLSIdx_.end(); ++p2) {
      // Check if this index is valid
      if (*p != -1) {
        // Check if any of the residuals are larger than the tolerance.
        if (testvector_[ *p2 ] > tolerance_) {
          // do nothing.
        } else if (testvector_[ *p2 ] == Teuchos::ScalarTraits<Scalar>::magnitude(Teuchos::ScalarTraits<Scalar>::zero())) {
          reset();
        } else if (testvector_[ *p2 ] <= tolerance_) {
          ind_[have] = *p2;
          have++;
        } else {
          // Throw an std::exception if a NaN is found.
          status_ = Failed;
          TEUCHOS_TEST_FOR_EXCEPTION(true,StatusTestError,"StatusTestGenResSubNorm::checkStatus(): NaN has been detected.");
        }
      }
    }
```
It seems that the seg fault is due to `testvector_[ *p2 ]` and that `*p2` is equal to `-1`.

I think that the test `if (*p != -1)` should be replaced by `if (*p2 != -1)` as [BelosXpetraStatusTestGenResSubNorm.hpp](https://github.com/trilinos/Trilinos/blob/develop/packages/muelu/adapters/belos/BelosXpetraStatusTestGenResSubNorm.hpp) is a template specialization of [BelosStatusTestGenResSubNorm.hpp](https://github.com/trilinos/Trilinos/blob/develop/packages/belos/src/BelosStatusTestGenResSubNorm.hpp) where the test is indeed `if (*p2 != -1)`.

**Comment by kliegeois** [comment]
My bad, the seg fault was coming from another instruction. 
Nevertheless, I think that `if (*p != -1)` is potentially a typing mistake.

**Comment by kliegeois** [comment]
Resolved by #6834 .
